python_modules/libraries/dagster-aws/dagster_aws/pipes.py

```diff
@@ -1,16 +1,58 @@
+import json
+import os
 import random
 import string
 from contextlib import contextmanager
-from typing import Iterator, Optional
+from typing import TYPE_CHECKING, Iterator, Optional
 
 import boto3
 import dagster._check as check
 from botocore.exceptions import ClientError
 from dagster._core.pipes.client import (
+    PipesContextInjector,
     PipesParams,
 )
 from dagster._core.pipes.utils import PipesBlobStoreMessageReader, PipesBlobStoreStdioReader
 
+if TYPE_CHECKING:
+    from dagster_pipes import PipesContextData
+
+_CONTEXT_FILENAME = "context.json"
+
+
+class PipesS3ContextInjector(PipesContextInjector):
+    """A context injector that injects context by writing to a temporary S3 location.
+
+    Args:
+        bucket (str): The S3 bucket to write to.
+        client (boto3.client): A boto3 client to use to write to S3.
+        key_prefix (Optional[str]): An optional prefix to use for the S3 key. Defaults to a random
+            string.
+
+    """
+
+    def __init__(self, *, bucket: str, client: boto3.client):
+        super().__init__()
+        self.bucket = check.str_param(bucket, "bucket")
+        self.key_prefix = "".join(random.choices(string.ascii_letters, k=30))
```

Comment:
dynamic key prefix should be per invocation right? If i re-use the same injector instance via the same resource instance for multiple steps we will collide on keys. I think currently this is not a problem since that only happens in serial in process execution but  probably good to defend for future in process concurrency